common/src/main/java/com/wynntils/functions/StatisticFunctions.java

```diff
@@ -0,0 +1,150 @@
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return DEFAULT_VALUE;
+
+            return statistic.count();
+        }
+    }
+
+    public static class StatisticsMinFunction extends StatisticFunction<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return DEFAULT_VALUE;
+
+            return statistic.min();
+        }
+    }
+
+    public static class StatisticsMaxFunction extends StatisticFunction<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return DEFAULT_VALUE;
+
+            return statistic.max();
+        }
+    }
+
+    public static class StatisticsAverageFunction extends StatisticFunction<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return DEFAULT_VALUE;
+
+            return statistic.average();
+        }
+    }
+
+    public static class StatisticsFirstModifiedFunction extends StatisticFunction<Long> {
+        @Override
+        public Long getValue(FunctionArguments arguments) {
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return 0L;
+
+            return statistic.firstModified();
+        }
+    }
+
+    public static class StatisticsLastModifiedFunction extends StatisticFunction<Long> {
+        @Override
+        public Long getValue(FunctionArguments arguments) {
+            StatisticEntry statistic = getStatisticEntry(arguments);
+            if (statistic == null) return 0L;
+
+            return statistic.lastModified();
+        }
+    }
+
+    public static class StatisticsFormattedFunction extends Function<String> {
+        @Override
+        public String getValue(FunctionArguments arguments) {
+            String statisticKey = arguments.getArgument("statisticKey").getStringValue();
+            int value = arguments.getArgument("value").getIntegerValue();
+
+            StatisticKind statisticKind;
+            try {
+                statisticKind = StatisticKind.from(statisticKey);
+            } catch (IllegalArgumentException e) {
+                return "-";
+            }
+
+            return statisticKind.getFormattedValue(value);
+        }
+
+        @Override
+        public FunctionArguments.Builder getArgumentsBuilder() {
+            return new FunctionArguments.RequiredArgumentBuilder(List.of(
+                    new FunctionArguments.Argument<>("statisticKey", String.class, null),
+                    new FunctionArguments.Argument<>("value", Number.class, null)));
+        }
```

Comment:
I'm a little confused on why this function needs these 2 arguments, I think you meant so that you can pass in the character or overall value for the 2nd argument, but wouldn't it be simpler to have a boolean for character or overall?

Reply:
It's not meant to be a boolean. It's meant to be a numeric value to be formatted. Each statistic key has a it's own formatter.

![image](https://github.com/Wynntils/Artemis/assets/49001742/a5b50003-c9d6-4569-be56-77d7f48d648f)

Reply:
Ah yeah that makes sense now